OpenDreamShared/Compiler/DMPreprocessor/DMPreprocessorLexer.cs

```diff
@@ -293,7 +293,7 @@ class DMPreprocessorLexer : TextLexer {
                             if (!error) token = CreateToken(TokenType.DM_Preproc_Number, textBuilder.ToString());
                             else token = CreateToken(TokenType.Error, null, "Invalid number");
                         } else {
-                            Advance();
+                            token = CreateToken(TokenType.Error, null, $"Unknown character: {c.ToString()}");
```

Comment:
This needs to move on from the unknown token, or it will get stuck in an infinite loop.
```suggestion
                            Advance();
                            token = CreateToken(TokenType.Error, null, $"Unknown character: {c.ToString()}");
```